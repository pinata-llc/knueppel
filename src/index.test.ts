import { buildQuery } from "./index";
import { getMockResolver } from "./utils/test_helpers";

// tslint:disable-next-line
const testJson = require("./index.test.json");

it("should build query object from a JSON AST", async () => {
  const identifiers = {
    identifier1: {
      queryString: "coalesc(table2.columnB, table3.columnC, ?)",
      bindings: [10],
      tables: ["table2, table3"],
    },
    identifier2: {
      queryString: "(select columnX from tableX where tableX.columnA = table1.columnA)",
      tables: ["table1"],
    },
  };

  const query = await buildQuery(testJson, getMockResolver(identifiers));
  expect(query).toMatchSnapshot();
  expect(query.compile()).toMatchSnapshot();
});
