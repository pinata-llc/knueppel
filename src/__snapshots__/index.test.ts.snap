// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should build query object from a JSON AST 1`] = `
Query {
  "bindings": Array [
    Query {
      "bindings": Array [
        Query {
          "bindings": Array [
            Query {
              "bindings": Array [
                "a",
              ],
              "queryString": "?",
              "tables": undefined,
            },
            Query {
              "bindings": Array [
                Query {
                  "bindings": Array [
                    "a",
                    "b",
                    "c",
                  ],
                  "queryString": "array[?,?,?]::text[]",
                  "tables": undefined,
                },
              ],
              "queryString": "any(?)",
              "tables": undefined,
            },
          ],
          "queryString": "(? = ?)",
          "tables": undefined,
        },
        Query {
          "bindings": Array [
            Query {
              "bindings": Array [
                false,
              ],
              "queryString": "?",
              "tables": undefined,
            },
          ],
          "queryString": "not (?)",
          "tables": undefined,
        },
      ],
      "queryString": "(? AND ?)",
      "tables": undefined,
    },
    Query {
      "bindings": Array [
        Query {
          "bindings": Array [
            10,
          ],
          "queryString": "coalesc(table2.columnB, table3.columnC, ?)",
          "tables": Array [
            "table2, table3",
          ],
        },
        Query {
          "bindings": Array [],
          "queryString": "(select columnX from tableX where tableX.columnA = table1.columnA)",
          "tables": Array [
            "table1",
          ],
        },
      ],
      "queryString": "(? in ?)",
      "tables": undefined,
    },
  ],
  "queryString": "(? OR ?)",
  "tables": undefined,
}
`;

exports[`should build query object from a JSON AST 2`] = `
Object {
  "bindings": Array [
    "a",
    "a",
    "b",
    "c",
    false,
    10,
  ],
  "queryString": "(((? = any(array[?,?,?]::text[])) AND not (?)) OR (coalesc(table2.columnB, table3.columnC, ?) in (select columnX from tableX where tableX.columnA = table1.columnA)))",
  "tables": Array [
    "table2, table3",
    "table1",
  ],
}
`;
